### ISSUE TITLE
bullets are applied to non-selected line

### ISSUE THREAD

**Issue by Alexey-T** [tomboy-notes/tomboy-ng#264]
I select 2 lines: 
```
[aa
bb
]cc
```
from start of line 'aa' to start of line 'cc'. it means line 'cc' is not selected at all. but 'bullets' are applied to all 3 lines. it must not be so - they must apply only to 'aa' and 'bb'.

![tom](https://user-images.githubusercontent.com/4154978/174097390-1a5a466d-4ad5-42de-8a35-7a25a6b78208.png)

**Comment by davidbannon** [comment]
This and the other issue about bullets have the same answer.

Sadly, KMemo does not do bullets well. So, my code does a lot of works before passing user input through to Kmemo whenever bullets might be involved. The code is particularly ugly, I suspect slows down input and is an absolute nightmare to debug. TK has indicated he is not interested in further developing KControls (but has accepted a few of my fixes).
One of the problems, IMHO, is that in KMemo, the attributes of a block of text are imbedded at the end of the text, not the beginning. Its intrinsic to KMemo and is not going to change.

So, in summary, while I agree its not perfect, the thought of digging back into that code does not excite me. I have, I estimate, several thousand users (*) and no one else has reported this matter and any further pre-processing will further slow down the response to each key press.

(*) Based on just direct downloads following a new release. I have no way of estimating how many users install direct from their distro repository.

But, that being said, a pull request would obviously be looked at favorably, have a look at code for editbox.pas, line 2715 on. There is quite a bit going on ....

Davo

**Comment by davidbannon** [comment]
FYI : Much of tomboy-ng's bullet code has just been rewritten. Won't be released for some time due to need for lots of testing of such a drastic change. But it may well have solved this problem.
Davo
